### DISCUSSION TITLE
gltf-transform etc1s skipping normal texture compression

### DISCUSSION THREAD

**Discussion by lanxinger** [donmccurdy/glTF-Transform#625] [Q&A] [answered]
When testing the etc1s compression I noticed that it skips the normal texture everytime. can I force it to also compress the normal texture?

```
debug: toktx:texture(baked_mesh_norm0): 17.47 MB → 17.47 MB bytes
debug: toktx:texture(baked_mesh_tex0): Spawning → toktx --genmipmap --bcmp --threads 32 C:\Users\marku\AppData\Local\Temp\tmp-19952-YMzv4bSFs4ka-.ktx2 C:\Users\marku\AppData\Local\Temp\tmp-19952-E6Z1vc4YHH8Z-.png
debug: toktx:texture(baked_mesh_tex0): 25.93 MB → 2.64 MB bytes
debug: unpartition: Complete.
info: untitled.glb (43.76 MB) → etc1s.glb (22.5 MB)
```

**Comment by donmccurdy** [comment]
Hm, it [shouldn't be printing that output](https://github.com/donmccurdy/glTF-Transform/blob/main/packages/cli/src/transforms/toktx.ts#L194) if it's skipping the file. This doesn't happen on the models I'm testing here, but could be a bug – would you be able to share the file?

**Comment by lanxinger** [comment]
Thanks for your quick answer. 

Here is the file:
https://1drv.ms/u/s!Aqb0QPUVo9J7hMg0bbzSSaEvgxwwtg?e=3Fw8T8

**Comment by lanxinger** [comment]
@donmccurdy I also tested this with a Khronos glTF sample model and observed the same happening there. it converts all textures to ktx2 except the normal map. 

Test with this file:
[BoomBox.glb.zip](https://github.com/donmccurdy/glTF-Transform/files/9127249/BoomBox.glb.zip)

**Comment by donmccurdy** [comment, marked as answer]
This should be fixed by https://github.com/donmccurdy/glTF-Transform/pull/635, in v2.2.2. The cause appears to have been an update in the KTX-Software CLI, changing the name of a normal-map-related parameter.

**Comment by lanxinger** [comment]
thank you @donmccurdy . 

however after testing I found that now the normal map is getting changed to monochrome. I have attached the result file for you.

[out.glb.zip](https://github.com/donmccurdy/glTF-Transform/files/9173165/out.glb.zip)

**Reply by donmccurdy** [reply]
Hm, the new version of KTX-Software does not do what I thought it did. I don't know how to make this work in KTX-Software v4.1.0-rc1, and have filed an issue about it at https://github.com/KhronosGroup/KTX-Software/issues/600. In the meantime you may need to revert to KTX-Software v4.0.0 to get this to work.

- /cc https://github.com/donmccurdy/glTF-Transform/issues/636

**Reply by donmccurdy** [reply]
It is also probably worth pointing out that ETC1S compression does not tend to work very well for normal maps. You may need to use UASTC, or keep the normal maps in PNG, for best quality. More details here:

https://github.com/KhronosGroup/3D-Formats-Guidelines/blob/main/KTXArtistGuide.md

**Reply by donmccurdy** [reply]
Just made another release as v2.2.3 with what should be the last round of fixes here!

**Comment by lanxinger** [comment]
thank you @donmccurdy. that worked. really appreciate the quick fixes and your tool in general.
